proxy/context.go

```diff
@@ -20,10 +20,13 @@ const unknownHost = "_unknownhost_"
 type flushedResponseWriter interface {
 	http.ResponseWriter
 	http.Flusher
+	SetReadDeadline(time.Time) error
+	SetWriteDeadline(time.Time) error
```

Comment:
Why do we need these?

Reply:
to set the corresponding deadlines

Reply:
I think we would not need it if we implement proper Unwrap method

Reply:
done